src/components/elements/metadataKeyValue/MetadataKeyValuePair.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import Box from '@mui/material/Box';
+import CustomTextField from './CustomTextField';
+import { DropdownV2 } from '../dropdownV2';
+import { RadioButtonItemProps } from '../../forms';
+import DeleteOutlinedIcon from '@mui/icons-material/DeleteOutlined';
+import IconButton from '@mui/material/IconButton';
+
+export interface MetadataKeyValuePairProps {
+  index: number;
+  keyName?: string;
+  value?: string;
+  valueType?: string;
+  isEditable: boolean;
+  isAutoGenerated?: boolean;
+  isNewlyCreated?: boolean;
+  deletionEnabled?: boolean;
+  saveKeyNameLocally?: (index: number, newVal?: string | undefined) => void;
+  saveValueLocally?: (index: number, newVal?: string | undefined) => void;
+  saveValueTypeLocally?: (index: number, newVal?: string | number | undefined) => void;
+  deleteFieldPermanently?: (index: number) => void;
+}
+
+export function MetadataKeyValuePair({
+  index,
+  keyName,
+  value,
+  valueType,
+  isEditable,
+  isAutoGenerated,
+  isNewlyCreated,
+  deletionEnabled,
+  saveKeyNameLocally,
+  saveValueLocally,
+  saveValueTypeLocally,
+  deleteFieldPermanently
+}: MetadataKeyValuePairProps) {
+  const valueTypes: RadioButtonItemProps[] = [
+    {
+      id: 'INTEGER',
+      label: 'Int'
+    },
+    {
+      id: 'FLOAT',
+      label: 'Float'
+    },
+    {
+      id: 'BOOLEAN',
+      label: 'Boolean'
+    },
+    {
+      id: 'STRING',
+      label: 'String'
+    },
+    {
+      id: 'BLOB',
+      label: 'Blob'
+    }
+  ];
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        width: '100%',
+        height: '100%',
+        backgroundColor: '#F8FAFC',
+        borderBottom: '1px solid #E2E8F0',
+        alignItems: 'center',
+        gap: '8px'
+      }}
+    >
+      <Box
+        sx={{
+          display: 'flex',
+          width: '35%',
+          height: '100%',
+          flexShrink: 0
+        }}
+      >
+        {/*key name should not be editable unless its newly created*/}
+        <CustomTextField
+          readOnly={!isNewlyCreated}
+          value={keyName}
+          saveLocallyHandler={(newVal) => {
+            if (saveKeyNameLocally) {
+              saveKeyNameLocally(index, newVal);
+            }
+          }}
+          helperText={isAutoGenerated ? 'Auto-generated' : undefined}
```

Comment:
I think this should be more generic. The parent should be able to send a "subtext" or "description". It can be simple text or a component. Then "Auto-generated" is passed by the parent as a prop.